### PR TITLE
Testing pullrequest

### PR DESCRIPTION
**Pull request by ROMELIKE**

### CODE REVIEW

File: app/Http/Controllers/MemberController.php [hunk 1]
```diff
@@ -98,9 +97,9 @@ public function postAdd(MemberRequest $request)
         }
         $member->created_at = date('y-m-d H:i:s');
 
-
         //Handle Add member:
         $action = $model->addMember($member);
+        dd($action->message);
```
Comment (quanghoan) [inline app/Http/Controllers/MemberController.php]: pls remove dd();

File: public/admin/js/AngularJS/main.js [hunk 2]
```diff
@@ -106,7 +107,7 @@ MyApp.controller('MyController', ['$scope', '$http', '$log', 'Upload', '$timeout
                 $("#myModal").modal('hide');
 
             }, function errorCallback(response) {
-                console.log(response.data);
+                console.log(response);
```
Comment (quanghoan) [inline public/admin/js/AngularJS/main.js]: pls remove none-fuction code such as console.log(), dd(), var_dump()

File: app/Http/Controllers/MemberController.php [hunk 3]
```diff
@@ -55,13 +54,15 @@ public function postAdd(MemberRequest $request)
...
-
-            //validate extension of upload image:
-            $ext = $request->avatar->getMimeType();
-            if (in_array($ext,
-                ['image/jpeg', 'image/png', 'image/jpg', 'gif'])) {
+            $mimeType = $request->avatar->getClientMimeType();
+            $extAddition = $request->avatar->getClientOriginalExtension();
+            if (in_array($mimeType,
+                    ['image/jpeg', 'image/png', 'image/jpg', 'gif','inode/x-empty'])
+                AND in_array($extAddition, ['png', 'jpg', 'jpeg', 'gif'])
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php]: Please use && instead of AND

File: app/Http/Controllers/MemberController.php [hunk 4]
```diff
@@ -55,13 +54,15 @@ public function postAdd(MemberRequest $request)
...
+            ) {
                 //validate capacity of upload image:
                 if (!($request->avatar->getSize() > 10485760)) {
+//                    dd($request->avatar->getSize());
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php]: Please remove this line.

File: app/Http/Controllers/MemberController.php [hunk 5]
```diff
@@ -76,10 +77,11 @@ public function postAdd(MemberRequest $request)
                     //create new picture.
                     $newThumbnail = $newName.".".$extension;
 
-                    //move image to appropriate Folder:
-                    $request->avatar->move('admin/images/avatars/',
-                        $newThumbnail);
-
+                    if($mimeType!='inode/x-empty'){
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php]: This is not PSR-2, please run phpcs before pushing.

File: app/Http/Requests/MemberRequest.php [hunk 6]
```diff
@@ -33,14 +33,14 @@ public function rules()
...
-            'name.not_in'=>'Enter your name',
+            'name.not_in'=>'Enter your name ( < 100)',
             'name.required'=>'Enter your name',
             'name.max'=>'can not be greater than 100 characters',
             'age.numeric'=>'age must be a number',
             'age.digits_between:1,2'=>'age must only has 2 digits',
             'age.required'=>'Enter the age',
             'address.required'=>'address is nacessary',
-            'address.not_in'=>'address is nacessary',
+            'address.not_in'=>'address is nacessary ( < 300)',
```
Comment (4c0n) [inline app/Http/Requests/MemberRequest.php]: nacessary -> necessary

File: database/factories/ModelFactory.php [hunk 7]
```diff
@@ -12,7 +12,7 @@
 */
 
 /** @var \Illuminate\Database\Eloquent\Factory $factory */
-$factory->define(App\User::class, function (Faker\Generator $faker) {
+$factory->define(App\Model\User::class, function (Faker\Generator $faker) {
     static $password;
```
Comment (4c0n) [inline database/factories/ModelFactory.php:16]: Please do not use static variables.

File: public/admin/js/AngularJS/main.js [hunk 8]
```diff
@@ -86,6 +86,7 @@ MyApp.controller('MyController', ['$scope', '$http', '$log', 'Upload', '$timeout
                 age: $scope.member.age,
                 avatar: $scope.file,
             };
+            // console.log($scope.file);
```
Comment (4c0n) [inline public/admin/js/AngularJS/main.js:92]: Please remove console.log(), dd() and var_dump() lines. Debugging lines should not be committed.

File: tests/Feature/MemberControllerTest.php [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+class MemberControllerTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    /**
+     * Purpose: test view list member return.
+     */
+    public function testList()
+    {
+        $member = factory(MemberModel::class)->create();
```
Comment (4c0n) [inline tests/Feature/MemberControllerTest.php]: $member does not seem to be used. Please use it or remove it.

File: tests/Feature/WrongMemberControllerTest.php [hunk 10]
```diff
@@ -0,0 +1,369 @@
...
+    {
+        $newMember = [
+            'name' => '01234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567891',
+            'address' => 'NamTuLiem,HaNoi',
+            'age' => 23,
+            'email' => 'romecody@gmail.com',
+        ];
+
+        $response = $this->call('POST', route('post.add'), $newMember);
+        $this->assertEquals(302, $response->status());
```
Comment (4c0n) [inline tests/Feature/WrongMemberControllerTest.php:27]: If the controller also gives a JSON response, please assert the response message.

**Review by 4c0n — CHANGES_REQUESTED** [review]

**Pull request merged**